### PR TITLE
Minor refactoring of podutils.go

### PR DESCRIPTION
**Pull request by pswica**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Just a small refactor. I think it matches more of what I see in the codebase.
**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/util/podutils/podutils.go [hunk 1]
```diff
@@ -75,9 +78,9 @@ func getPodConditionFromList(conditions []corev1.PodCondition, conditionType cor
 	if conditions == nil {
 		return -1, nil
 	}
-	for i := range conditions {
-		if conditions[i].Type == conditionType {
-			return i, &conditions[i]
+	for i, condition := range conditions {
```
Comment (draveness) [inline pkg/kubectl/util/podutils/podutils.go]: the original code is intentional, FYI: https://github.com/golang/go/wiki/CommonMistakes

Comment (draveness) [reply pkg/kubectl/util/podutils/podutils.go]: `var` in for loop is the same variable which takes on all value of each slice element, though it does not cause any thouble here to return the address of for loop variable. But it is the correct approach to return `conditions[i]`

Comment (pswica) [reply pkg/kubectl/util/podutils/podutils.go]: I believe the link you sent refers to using variables within for loops that are scoped within goroutines. In a simple for loop here, I don't think it applies to simple for loops. 

For a simple for loop I tried to follow effective go: https://golang.org/doc/effective_go.html#for

```
 If you're looping over an array, slice, string, or map, or reading from a channel, a range clause can manage the loop.

for key, value := range oldMap {
    newMap[key] = value
}
```

**Pull request closed without merging**